base/inference.jl

```diff
@@ -5202,6 +5217,21 @@ function occurs_outside_getfield(e::ANY, sym::ANY,
         if head === :(=)
             return occurs_outside_getfield(e.args[2], sym, sv,
                                            field_count, field_names)
+        elseif head === :foreigncall
+            args = e.args
+            nccallargs = args[5]::Int
+            # Only arguments escape, GC root arguments do not escape.
```

Comment:
"escape" here refer to the structure/layout. If it's used as an argument then we do need to allocate memory to make it have the correct layout. If that use doesn't escape the pointer, it'll be the job of `llvm-alloc-opt.cpp` to remove the heap allocation.